OVERVIEW.md

```diff
@@ -0,0 +1,71 @@
+```
+# use bauble::Bauble;
+
+#[derive(Bauble)]
+struct TypeInBauble {
+   a_number: u32,
+   a_string: String,
+   a_bool: bool,
+}
+```
+
+More information on the [`Bauble`] derive macro exists in its documentation.
+
+# Getting started with using Bauble
+
+In order to use Bauble, the first step is to create a [`BaubleContext`]. The [`BaubleContext`] can be created via a [`BaubleContextBuilder`]. With the builder you may register various types and traits. The types and traits you provide to the builder can then be used to create values in Bauble which are associated with those types you registered from Rust. For example, if you create a struct in Rust that represents a person `struct Person { name: String, age: u32 }`, that struct can then be written out in Bauble like `person = Person { name: "Maria", age: 21 }`, or, `person: Person = Person { name: "David", age: 35 }`. In order to register a trait, the builder expects a type in the form of `dyn Trait`. More on traits in the overview of Bauble. The builder is also capable of doing various other things like registering preludes (implicitly used types) or a required trait for all types, amongst others. Note, in order for Bauble to recognize a type as deriving a specific trait, it has to be registered manually to the builder at this stage. Once you've registered all your types and traits to the builder, you can build the context.
+
+Once you have your context, it can be used to register Bauble source files using [`BaubleContext::register_file`] that will be parsed by the context when you invoke [`BaubleContext::reload_paths`]. When invoking [`reload_paths`](BaubleContext::reload_paths) it returns all Bauble 'objects' that were parsed, as well as any potential errors that occured when parsing the Bauble source. A Bauble object is a typed lexical element of Bauble and can be be converted into a value in Rust which implements [`Bauble`], however it depends on the implementation of the [`Bauble`] whether the conversion is successful. The [`Bauble`] trait has a method called [`from_bauble`](Bauble::from_bauble) which is what is used to convert from a Bauble value (which is contained within a Bauble object) into a concrete, typed value in Rust. In order to support all Bauble values being parsed and represented as a single type in Rust, using type erasure is a possibility, however it has to be manually implemented. For example, a single `ErasedBauble` type which can represent any type that implements `Bauble`, but has to be explicitly downcast to the concrete type sometime later at runtime (when the type is known).  
+
+## Bauble overview
+
+# Includes (use)
+
+All Bauble source files support the usage of `use`. Similar to Rust, it may include any item defined from a separate Bauble module, as long as both modules (the module being used, and the module using) exist within the same [`BaubleContext`]. Alternatively to `use`, like in Rust, the fully qualified path may be written out instead.
```

Comment:
we might want to note that this can be used both with assets (aka objects?) and registered types